modules/javafx.graphics/src/main/native-prism-d3d/D3DMeshView.cc

```diff
@@ -289,19 +289,22 @@ void D3DMeshView::render() {
         return;
     }
 
-    SUCCEEDED(device->SetTexture(SR_DIFFUSE_MAP, material->getMap(DIFFUSE)));
-    SUCCEEDED(device->SetTexture(SR_SPECULAR_MAP, material->getMap(SPECULAR)));
-    SUCCEEDED(device->SetTexture(SR_BUMPHEIGHT_MAP, material->getMap(BUMP)));
-    SUCCEEDED(device->SetTexture(SR_SELFILLUM_MAP, material->getMap(SELFILLUMINATION)));
+    // Set the textures and their filters in their corresponding samplers
+    for (int i = 0; i < map_type::num_map_types; i++) {
+        map_type type = static_cast<map_type>(i);
+        SUCCEEDED(device->SetTexture(type, material->getMap(type)));
```

Comment:
Wrapping the calls with the `SUCCEEDED` macro doesn't do anything useful.

Reply:
Yes, but it's done everywhere else in the class, and was done for these calls before I changed this code. Maybe a better error handling design is needed.